tools/adios2pio-nm/adios2pio-nm-lib.cxx

```diff
@@ -839,10 +1078,14 @@ DimensionMap ProcessDimensions(IOVector &bpIO, EngineVector &bpReader, int ncid,
                 { \
                     ierr = BP2PIO_ERROR; \
                 } \
-                if (ierr == BP2PIO_NOERR) { \
+                if (ierr == BP2PIO_NOERR) \
+                { \
                     int dimid; \
                     PIO_Offset *d_val = (PIO_Offset*)dimval.data(); \
+                    PIOc_redef(ncid); \
                     ret = PIOc_def_dim(ncid, dimname.c_str(), *d_val, &dimid); \
```

Comment:
Need to handle the return value from PIOc_def_dim()

Reply:
I think you mean PIOc_redef()? I will handle the return value.

Reply:
I think you need to handle return values from both the functions

Reply:
Current code only handles PIOc_def_dim
It should handle all of the 3 SCORPIO API calls: PIOc_redef, PIOc_def_dim and PIOc_enddef